frontend/catalyst/python_interface/visualization/construct_circuit_dag.py

```diff
@@ -69,6 +69,72 @@ def _visit_block(self, block: Block) -> None:
         for op in block.ops:
             self._visit_operation(op)
 
+    # =============
+    # CONTROL FLOW
+    # =============
+
+    @_visit_operation.register
+    def _for_op(self, operation: scf.ForOp) -> None:
+        """Handle an xDSL ForOp operation."""
+        cluster_id = f"cluster_{id(operation)}"
+        self.dag_builder.add_cluster(
+            cluster_id,
+            node_label="for ...",
+            cluster_id=self._cluster_stack[-1],
+        )
+        self._cluster_stack.append(cluster_id)
```

Comment:
Let me know what you think, but the `_cluster_stack` seems like something that should be part of the DAGBuilder

Reply:
Honestly, the more I think about it, it doesn't really matter where it lives. I'll leave it up to your discretion, but with [this suggestion](https://github.com/PennyLaneAI/catalyst/pull/2214/files#r2566148503), it might make sense to just keep it in the director class.